aleph0/src/createActions.ts

```diff
@@ -25,20 +42,45 @@ export const createActions = (
         properties: {},
       },
     },
-    readSnippetFile: {
-      function: async (args: { filePath: string }) => {
-        return { fileContents: readFile(args.filePath) };
+    expandSnippet: {
+      function: async (args: { id: string }) => {
+        const { file, references } = files[args.id];
+
+        const directory = process.cwd();
+
+        const filePath = `${directory}/${file}`;
+        const fileContents = readFile(filePath);
+        const referenceContents = references?.map((reference) => {
+          return {
+            name: reference,
+            contents: readFile(`${directory}/${reference}`),
+          };
+        });
+
+        const contents = `## File:
+${fileContents}
+
+## References:
+
+${referenceContents?.map(
+  (reference) => `### ${reference.name}
+${reference.contents}`
+)}
+`;
+
+        return contents;
       },
-      name: "readSnippetFile",
-      description: "Returns the contents of a snippet file.",
+      name: "expandSnippet",
+      description: "Returns the contents and references of a snippet file.",
       parse: (args: string) => {
-        return z.object({ filePath: z.string() }).parse(JSON.parse(args));
+        return z.object({ id: z.string() }).parse(JSON.parse(args));
       },
       parameters: {
         type: "object",
         properties: {
-          filePath: {
+          id: {
             type: "string",
+            description: "The id of the snippet file.",
```

Comment:
Give it an example so it will make less mistakes?

Reply:
Haven't seen it make a mistake here. But also not sure what example to give as it's dynamic and depends on the snippet. I'd leave as is.

Reply:
I mean the format of the kebab case and such with an (eg. Looks like ...)